src/Features/Core/Portable/CodeFixes/AddImport/AbstractAddImportCodeFixProvider.SearchScope.cs

```diff
@@ -112,11 +112,24 @@ public override SymbolReference CreateReference<T>(SearchResult<T> searchResult)
                     _metadataReference);
             }
 
-            protected override Task<IEnumerable<ISymbol>> FindDeclarationsAsync(string name, SymbolFilter filter, SearchQuery searchQuery)
+            protected override async Task<IEnumerable<ISymbol>> FindDeclarationsAsync(string name, SymbolFilter filter, SearchQuery searchQuery)
             {
-                return SymbolFinder.FindDeclarationsAsync(
+                // when searching metadata, give ourselves a limit on how long the search can take.
+                // That way AddImport doesn't pause for a long time while producing metadata indices.
+                const int SearchTimeoutMS = 50;
+
+                var timeoutTask = Task.Delay(millisecondsDelay: SearchTimeoutMS, cancellationToken: cancellationToken);
+                var findTask = SymbolFinder.FindDeclarationsAsync(
                     _solution, _assembly, _metadataReference, searchQuery, filter, cancellationToken);
+
+                var completedTask = await Task.WhenAny(timeoutTask, findTask).ConfigureAwait(false);
+                if (completedTask == timeoutTask)
+                {
+                    return SpecializedCollections.EmptyEnumerable<ISymbol>();
```

Comment:
So I thought by the fancy state stuff in the other file the goal was to return what you did find within the timeout. But it looks like that's not the case here. What's the goal? I had _thought_ the idea was the cancellation gets sent via the cancellationToken, but clearly that's not the case? And we're just tossing whatever was found?